internal/providercache/package_install.go

```diff
@@ -26,6 +28,39 @@ import (
 // specific protocol and set of expectations.)
 var unzip = getter.ZipDecompressor{}
 
+const (
+	// httpClientRetryCountEnvName is the environment variable name used to customize
+	// the HTTP retry count for module downloads.
+	httpClientRetryCountEnvName = "TF_PROVIDER_DOWNLOAD_RETRY"
+
+	defaultRetry = 2
+)
+
+func init() {
+	configureDiscoveryRetry()
+}
+
+var (
+	maxRetryCount int
+)
+
+// will attempt for requests with retryable errors, like 502 status codes
+func configureDiscoveryRetry() {
```

Comment:
This area of code is for provider downloads, not discovery.

```suggestion
func configureProviderDownloadRetry() {
```